docs/extending.md

```diff
@@ -33,3 +35,34 @@ Make sure that you also define:
 
 Note that this "extension" mechanism is still experimental. If anything is not clear, or
 doesn't work, please do raise an issue or contact us on Discord (see the link on the README).
+
+## Creating a Plugin
+
+Another option is to write a plugin. Narwhals itself has the necessary utilities to detect and handle 
+plugins. For this integration to work, any plugin architecture must contain the following:
+
+  1. an entrypoint defined in a `pyproject.toml` file:
+
+    ```
+    [project.entry-points.'narwhals.plugins']
+    narwhals-<library name> = 'narwhals_<library name>'
+    ```
+    The first line needs to be the same for all plugins, whereas the second is to be adapted to the 
+    library name.
```

Comment:
this may be confusing, as technically `[project.entry-points` would be the section and its first line would be `narwhals-<library name`

perhaps just write that the section name needs to be the same for all plugins, and they can replace their own library name inside it, for example `narwhals-grizzlies = 'narwhals_grizzlies'`